src/app/app.component.ts

```diff
@@ -66,6 +80,7 @@ export class AppComponent implements OnInit {
     if(this.snake.getHead() === this.grid.getOutOfBoundsCell()){
         return true;
     }
-    return false;
+    
+    return this.walls.some(wall => wall.hasColision(this.snake));
```

Comment:
Might be better if we just return both this and the above combined? Like remove the if and have
`return this.snake.getHead() === this.grid.getOutOfBoundsCell() ||  this.walls.some(wall => wall.hasColision(this.snake));`